\subsection{Artifacts in Genomic Datasets}

% \2 Background \& Motivation
% 	\3 Artifacts in genomic datasets are common, notably frameshifts and
% 		early stop codons.
% 	\3 Artifacts negatively impact downstream analyses.
% 	\3 Recap of chapter 1?
% \2 Artifacts and ambiguous data
% 	\3 Analyze existing databases and plot distribution of artifacts
% 		(frameshifts \& early stop codons) and ambiguous nucleotides
% 		(has this been done before?).
% 	\3 Introduce IUPAC ambiguous nucleotide support into marginal COATi
% 		model.
% 		\4 Theory and implementation for descendant sequence.
% 		\4 Theory for reference sequence, implementation subject to time.
% \2 Model sequencing error
% 	\3 Sequencing FST
% 		\4 Gap unit size: 1 vs 3 vs 1\&3
% 		\4 Incorporate base-calling error profile?
% 	\3 How to treat ambiguous nucleotides.
% 		\4 Average vs most probable nucleotide (“best”).
% 	\3 Padding
% \2 Biological frameshifts
% \2 Validation
