\section{Introduction}
\green{TODO:
\begin{itemize}
    \item Split comps significance appropriately into intro and background.
    \item Add more information about artifacts (frameshifts and early stop cods).
\end{itemize}
}

\blue{
In genomics and bioinformatics, studies often start with the alignment of coding sequences.
An alignment is a hypothesis of which characters from two or more sequences are homologous
\parencite{problems_cartwright_2009} and their reconstruction is the most fundamental
computational task in bioinformatics \parencite{sequence_alignment_rosenberg_2009}.
Sequence alignment is also essential to other analyses, including the identification of conserved
motifs, estimation of evolutionary divergence between sequences, inference of phylogenetic
relationships \parencite{MSA_kumar_2007}, identification of disease-associated mutations, measurement
of selection, among others \parencite{sequence_alignment_rosenberg_2009}.}

% about programs still not fully addressing sequence homology and it being unique & unobservable historical events.
\blue{
Modern sequence analysis began with the heuristic homology algorithms of Needleman and Wunsch in 1970
\parencite{identification_smith_1981} and numerous others have been developed to arrive at current
aligners.
Despite extensive investments, in practice, sequence alignment is often merely seen as a tool and
the alignment inference as an ad hoc problem \parencite{morrison_MSA_2018}.
Inference of homology is an inherently difficult problem since the objective is backtracking unique and
unobservable historical events \parencite{sequence_aln_morrison_2010}.
Errors introduced in sequence alignment can, if not corrected, lead to erroneous phylogenetic inference,
detection of positive selection, protein structure prediction \parencite{metrics_blackburne_whelan_2011},
and other problems in comparative and functional genomic studies \parencite{estimates_schneider_2009}.
}

\section{Background}
